new-log-viewer/src/services/decoders/JsonlDecoder.ts

```diff
@@ -0,0 +1,240 @@
+        }
+        const dayjsObj: dayjs.Dayjs = dayjs.utc(timestamp);
+        const formattedDate = dayjsObj.format(this.#dateFormat);
+        input = input.replace(this.#datePattern, formattedDate);
+
+        return [
+            dayjsObj.valueOf(),
+            input,
+        ];
+    }
+
+    /**
+     * Replaces placeholders in the input string with corresponding properties from the logEvent
+     * object.
+     *
+     * @param input The input string with placeholders to be replaced.
+     * @param logEvent The log event object containing properties to replace the placeholders.
+     * @return - The input string with placeholders replaced by corresponding property values.
+     * @private
+     */
+    #formatVariables (input: string, logEvent: JsonObject) {
+        // Replace each placeholder with the corresponding property from logEvent
+        for (const propName of this.#propertyNames) {
+            if (propName in logEvent) {
+                const placeholder = `%${propName}`;
+                const propValue = logEvent[propName];
+                let propValueString: string;
+                if ("undefined" === typeof propValue || null === propValue) {
+                    console.error(`Unexpected undefined logEvent[${propName}]`);
+                    propValueString = "undefined";
+                } else if ("object" === typeof propValue) {
+                    propValueString = JSON.stringify(propValue);
+                } else {
+                    propValueString = propValue.toString();
+                }
+                input = input.replace(placeholder, propValueString);
+            }
+        }
+
+        input += "\n";
+
+        return input;
+    }
+
+    /**
+     * Extracts the verbosity from the given log event.
+     *
+     * @param logEvent The log event containing the verbosity.
+     * @return The extracted verbosity.
+     */
+    #extractVerbosity (logEvent: JsonObject) {
+        let verbosity = LOG_VERBOSITY.NONE;
+        const verbosityString: string = logEvent[this.#verbosityPropName] as string;
+        if (false === (verbosityString in LOG_VERBOSITY)) {
+            console.error(`Unable to find verbosity from key ${this.#verbosityPropName}` +
+                ` of type ${typeof verbosityString}`);
+        } else {
+            verbosity = LOG_VERBOSITY[verbosityString as (keyof typeof LOG_VERBOSITY)];
+        }
+
+        return verbosity;
+    }
+
+    constructor (dataArray: Uint8Array) {
+        this.#dataArray = dataArray;
+    }
+
+    setDecodeOptions (options: JsonlDecodeOptionsType): boolean {
+        this.#setTextPattern(
+            options.textPattern
+        );
+        this.#timestampPropName = options.timestampPropName;
+        this.#verbosityPropName = options.verbosityPropName;
+
+        return true;
+    }
+
+    buildIdx (): number {
+        const text = JsonlDecoder.TEXT_DECODER.decode(this.#dataArray);
+        const split = text.split("\n");
```

Comment:
```suggestion
    /**
    * Decodes log events from the #logEvents array and adds them to the results array.
    *
    * @param {DecodeResultType[]} results - The array to which the decoded log events will be added.
    * @param {number} startIdx - The index in the #logEvents array at which to start decoding.
    * @param {number} endIdx - The index in the #logEvents array at which to stop decoding.
    * @return {boolean} - Returns true if the decoding was successful, false otherwise.
    */
    decode (results: DecodeResultType[], startIdx: number, endIdx: number): boolean {
```